packages/grid/x-grid/src/tests/rows.XGrid.test.tsx

```diff
@@ -449,6 +449,35 @@ describe('<XGrid /> - Rows', () => {
     });
 
     describe('scrollToIndexes', () => {
+      it('should scroll correctly to the previous row of the first visible row', () => {
+        const headerHeight = 50;
+        const rowHeight = 50;
+        const border = 1;
+        const rowsToFullyDisplay = 4;
+        const viewportHeight = (rowsToFullyDisplay + 0.5) * rowHeight;
+
+        // The height where virtualization changes from one page to another
+        const renderingZoneHeight = rowsToFullyDisplay * 2 * rowHeight - viewportHeight;
+
+        render(
+          <TestCaseVirtualization
+            hideFooter
+            height={viewportHeight + headerHeight + border * 2}
+            headerHeight={headerHeight}
+            nbCols={2}
+            rowHeight={rowHeight}
+          />,
+        );
+        const gridWindow = document.querySelector('.MuiDataGrid-window')!;
+        // Scroll to the rowIndex=5, which is the 2nd row in page 1
+        gridWindow.scrollTop = renderingZoneHeight + rowHeight;
+        gridWindow.dispatchEvent(new Event('scroll'));
+        // Scroll to rowIndex=4, which is the 1st row in page 1
+        apiRef.current.scrollToIndexes({ rowIndex: 4, colIndex: 0 });
+        gridWindow.dispatchEvent(new Event('scroll'));
+        expect(gridWindow.scrollTop).to.equal(renderingZoneHeight);
```

Comment:
This doesn't look right. AFAIK, it should be
```suggestion
        expect(gridWindow.scrollTop).to.equal(4 * rowHeight);
```

We display the 5th row, so we need to scroll 4-row heights. Anything else would push us toward changing the scroll logic when it's the rendering logic that is wrong.

Reply:
#1969 doesn't solve the bug when the given `rowIndex` is way before the first visible row. This test is lying.

Reply:
Which bug are we talking about? If we are talking about #1818, I think that it should be put on hold until #1911 is fixed.

Reply:
You mentioned that this PR passes with the logic from #1969. From what I understood that means that we should only merge the test case here. I'm saying that the bug of #1818 remains and the test case is not right. I'll close this one and investigate how to address #1911.

Reply:
- :+1: for an extra test case. 
- #1911 I would expect the solution not to be too complex. I think that we should first look into why the scrollTop is too small compared to what it should be, then once fixed, look into why the rendered rows are not continuous. It's likely a small refactoring.

Reply:
@m4theushw Coming back to a discussion we had. Do you agree that this assertion would be correct? https://github.com/mui-org/material-ui-x/pull/1871/files#r660182225. If you do, I think that we could merge a new test case. If you don't, we can chat more about it.

Either way, what's the intended outcome of keeping the PR open for a couple of weeks? We could restart the effort from a blank branch.

Reply:
I do agree that https://github.com/mui-org/material-ui-x/pull/1871/files#r660182225 is the correct assertion, but it doesn't work this way. The virtualization changes from page 0 to 1 before that (at the "jump"). If we merge this one it will be too tied to the virtualization implementation. Better to close it and to fix the root cause.

Reply:
I was going to say that the value of this test case is to increase our test coverage, but I feel that it duplicate with https://github.com/mui-org/material-ui-x/blob/4066078512087318ec22e2b80763564d67b0a22e/packages/grid/x-grid/src/tests/rows.XGrid.test.tsx#L480

So, after all, I don't see any value in this PR 😁, outside of the discussion it has created.

Reply:
I'm testing it after rebasing with master. It seems to pass with the proposed assertion.

Reply:
Yeah, it passes but visually it's wrong.

https://user-images.githubusercontent.com/42154031/124522376-6eae1700-ddc9-11eb-8014-16c907d76211.mp4